src/scalar/fixed_size_list.rs

```diff
@@ -0,0 +1,69 @@
+use std::any::Any;
+use std::sync::Arc;
+
+use crate::{array::*, datatypes::DataType};
+
+use super::Scalar;
+
+/// The scalar equivalent of [`FixedSizeListArray`]. Like [`FixedSizeListArray`], this struct holds a dynamically-typed
+/// [`Array`]. The only difference is that this has only one element.
+#[derive(Debug, Clone)]
+pub struct FixedSizeListScalar {
+    values: Arc<dyn Array>,
+    is_valid: bool,
```

Comment:
What do you think about

```suggestion
    values: Option<Arc<dyn Array>>,
```

and remove `is_valid`?

Reply:
I referred to `ListScalar`, but I think `Option<Arc<dyn Array>>` is better and value method should return `Option<&Arc<dyn Array>>` for consistency of other scalar.

`StructScalar` and `ListScalar` also have `is_valid`. Should we use `Option` and remove `is_valid` for them? (Maybe another PR)